### ISSUE TITLE
Add softmax focal loss

### ISSUE THREAD

**Issue by luizfernando608** [pytorch/vision#8008]
### 🚀 The feature

Introduce the softmax version of focal loss as outlined in [Tsung-Yi Lin et al.'s 2018 paper, "Focal Loss for Dense Object Detection."](https://arxiv.org/abs/1708.02002)

### Motivation, pitch

I propose to add softmax focal loss to the repo as a new feature. Softmax focal loss is a variant of focal loss that can handle multi-class classification problems better than sigmoid focal loss, in some cases. 

Currently, we only have sigmoid focal loss.

I think it would be a valuable addition to our repo

**Comment by pmeier** [comment]
Duplicate of #3250.
